packages/grafana-ui/src/components/Link/TextLink.mdx

```diff
@@ -1,10 +1,152 @@
+
+1. [Usage](#usage)
+1. [Types](#types)
+1. [Behaviour](#behaviour)
+   <br />
+   4.1. [Inline](#inline-behaviour)
+   <br />
+   4.2. [Standalone](#standalone-behaviour)
+   <br />
+   4.3. [Icons](#icons-behaviour)
+1. [Props table](#propstable)
+1. [Related](#related)
+
+<br />
+<br />
+
+## <a name="usage"/> Usage
+
+<br />
+
+### **When to use**
+
+- To redirect the user to another page, external or internal
+- When redirecting the user through a link in a word or phrase within a paragraph.
+- When redirecting the user through a link in a word or phrase, not within a paragraph but surrounded by other blocks of text.
+
+### **When not to use**
+
+- When its purpose is to trigger an action, please use a Button or IconButton component
+- When it still redirects the user but it is not surrounded by text - for example, being in a row with more buttons -, please use the LinkButton component.
+- When the link is just comprised by an icon use IconButton component.
+
+### **Do’s**
+
+- Use the default styles as the first and main option as they are though to be accessible and consistent.
+- Use the external icon (the default one) when the TextLink goes to an external page as its meaning is broadly known.
+- In case the default icon does not fit, please use one of the icons of our [library](https://developers.grafana.com/ui/latest/index.html?path=/story/docs-overview-icon--icons-overview).
+
+### **Don’ts**
+
+- Do not use ‘primary’ color in a standalone link unless it is a special use case such as the ‘Dashboard list’, menu items or similar.
+- Do not override any styles in any way. Instead, ask the Design System team for an improvement of this component.
+- Do not attempt to add a new icon. You should fulfil a contribution path to first add the icon to our Design System and then use it in this component.
+
+  <br />
+  <br />
+
+## <a name="types"/> Types
+
+Within Grafana, there are two different types of _text links_, 'inline' or 'standalone', whether they are wrapped by text or being a standalone element.
+
+- **Inline**: It is part of a broader text. It will be surrounded by text, before, after or in both directions. Although it can show an icon at its end, we encouragedly advice not to do that.
+- **Standalone**: It is a block not surrounded immediately by more text.
+
+### **Subtypes**
+
+Both inline or standalone text links will redirect the user to a specific page, therefore, depending on this they can be internal or external.
+
+- **Internal**: Redirect the user to a page inside Grafana.
+- **External**: Redirect the user to a page outside Grafana. The page will open on a new window. By default, an external text link will show the ‘external-link-alt’ icon.
+
+  <br />
+  <br />
+
+## <a name="behaviour"/> Behaviour
+
+The following is the default behaviour and so, it will be applied according to its type.
+
+### <a name="inline-behaviour"/>**Inline**:<br/>
+
+<Preview>
+  <TextLink href="https://google.es" external inline>
```

Comment:
Both these previews look strange to me:

<img width="290" alt="Screenshot 2023-07-17 at 17 47 35" src="https://github.com/grafana/grafana/assets/100691367/9ed0e69e-1ce8-4824-8e46-dfab9388d3ca">

Reply:
I think this might be `Storybook` adding some kind of style to `Preview`. As you can see here, if I change something in the file they reload and look right 🤷‍♀️ 

![Captura de pantalla 2023-07-18 a las 13 02 39](https://github.com/grafana/grafana/assets/65417731/1cfcea6a-b340-48f7-890a-abfc5472280b)

Reply:
<img width="781" alt="Screenshot 2023-07-18 at 14 21 11" src="https://github.com/grafana/grafana/assets/100691367/2e9fb1e0-3c1b-4e89-ab34-69ab4fc35b65">

It seems to be because they are set to paragraph and these have some margin bottom set? I guess `Preview` sets all of them to `p`??